Setup CI test compensated-rails

user512 opened the pull request.

user512 committed:
Setup CI test compensated-rails

zspencer committed:
Merge branch '0.X' into setup-test-compensated-rails

zspencer committed:
Compensated Rails does not yet support Ruby 2.4

zspencer committed:
Grep stops execution because it fails with a 1

zspencer committed:
Doh, missed a spot

zspencer committed:
Tests work across Ruby versions

zspencer commented on compensated-rails/spec/dummy/config/application.rb:
```diff
@@ -8,7 +8,7 @@
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 6.0
+    config.load_defaults Rails.version.slice(0,3)
```
This allows us to support Rails 5.X~6.X; it may not be _the right_ way to do it, but I was doing something quick and dirty to get CI passing. Maybe we should look at how other Rails engines test across multiple Rails?

user512 replied:
A bit confused on how does it allow us to support Rails 5.X ~ 6.X?
This is related to changing autoloading mode?
Saw this from https://guides.rubyonrails.org/configuring.html#rails-general-configuration
> config.autoloader sets the autoloading mode. This option defaults to :zeitwerk if 6.0 is specified in config.load_defaults.

zspencer replied:
Without this change when we tried to include compensated-rails in a Rails 5 project, it would crash out because the value was hardcoded to `6.0` , which is not a valid option in Rails 5.

I'm not sure the _right_ thin to do here, but this at least gets stuff passing.

zspencer reviewed (approved):
I think this is a good enough patch; what do you think?

zspencer commented on compensated-rails/.gitignore:
```diff
@@ -7,3 +7,4 @@ spec/dummy/db/*.sqlite3-*
 spec/dummy/log/*.log
 spec/dummy/storage/
 spec/dummy/tmp/
+Gemfile.lock
```
Including the `Gemfile.lock` is a good idea in applications, but less so in gems where we want to test against multiple versions of Ruby/Gems.

user512 committed:
Merge branch '0.X' into setup-test-compensated-rails

zspencer committed:
Only install ruby in `bin/setup` for now.

There probably is a better way to ensure folks have ruby installed...
Not sure what it is yet... but maybe someday I will.

zspencer committed:
Missed a tiny part of bin/setup

zspencer committed:
Remove unnecessary branch

zspencer commented on compensated-rails/README.md:
```diff
@@ -25,4 +25,4 @@ $ gem install compensated-rails
 
 ## License
-The gem is available as open source under the terms of the [MIT License](https://opensource.org/licenses/MIT).
+While everyone has access to the source code, only non-commercial use is permitted. See [LICENSE.md for details](LICENSE.md).
```
Nice catch by the way.
